server/issue.go

```diff
@@ -49,42 +51,58 @@ func httpAPICreateIssue(ji Instance, w http.ResponseWriter, r *http.Request) (in
 		return http.StatusInternalServerError, err
 	}
 
-	// Lets add a permalink to the post in the Jira Description
-	post, appErr := api.GetPost(create.PostId)
-	if appErr != nil {
-		return http.StatusInternalServerError,
-			errors.WithMessage(appErr, "failed to load post "+create.PostId)
-	}
-	if post == nil {
-		return http.StatusInternalServerError,
-			errors.New("failed to load post " + create.PostId + ": not found")
-	}
-	if channel, _ := api.GetChannel(post.ChannelId); channel != nil {
-		if team, _ := api.GetTeam(channel.TeamId); team != nil {
-			permalink := fmt.Sprintf("%v/%v/pl/%v",
-				ji.GetPlugin().GetSiteURL(),
-				team.Name,
-				create.PostId,
-			)
-
-			if len(create.Fields.Description) > 0 {
-				create.Fields.Description += fmt.Sprintf("\n%v", permalink)
-			} else {
-				create.Fields.Description = permalink
+	var post *model.Post
+	var appErr *model.AppError
+
+	// If this issue is attached to a post, lets add a permalink to the post in the Jira Description
+	if create.AttachedToPost {
+		post, appErr = api.GetPost(create.PostId)
+		if appErr != nil {
+			return http.StatusInternalServerError,
+				errors.WithMessage(appErr, "failed to load post "+create.PostId)
+		}
+		if post == nil {
+			return http.StatusInternalServerError,
+				errors.New("failed to load post " + create.PostId + ": not found")
+		}
+		if channel, _ := api.GetChannel(post.ChannelId); channel != nil {
+			if team, _ := api.GetTeam(channel.TeamId); team != nil {
+				permalink := fmt.Sprintf("%v/%v/pl/%v",
+					ji.GetPlugin().GetSiteURL(),
+					team.Name,
+					create.PostId,
+				)
+
+				if len(create.Fields.Description) > 0 {
+					create.Fields.Description += fmt.Sprintf("\n%v", permalink)
+				} else {
+					create.Fields.Description = permalink
+				}
 			}
 		}
 	}
 
 	created, _, err := jiraClient.Issue.Create(&jira.Issue{
 		Fields: &create.Fields,
 	})
+
+	// For now, if we are not attaching to a post, just postId blank (this will only affect the error message)
+	var channelId, postId string
+	if create.AttachedToPost {
+		channelId = post.ChannelId
+		postId = create.PostId
+	} else {
+		channelId = create.ChannelId
+		postId = ""
+	}
+
 	if err != nil {
 		return http.StatusInternalServerError,
-			errors.WithMessage(err, "failed to create the issue, postId: "+create.PostId)
+			errors.WithMessage(err, "failed to create the issue, postId (if attached to a post): "+postId)
```

Comment:
nit: 1/5 I'd drop ` (if attached to a post)` from the error, `postId: nil` is intention-revealing enough.